lib/worldwide/address.rb

```diff
@@ -465,6 +465,33 @@ def strip_extra_japanese_chars(line:, excluded_fields:)
       end
     end
 
+    def script_from_string(str)
+      script = "default"
+      scripts = region.combined_address_format.except("default").keys
+      return script if scripts.empty?
+
+      detected_scripts = Worldwide::Scripts.identify(text: str).map(&:to_s)
+      if detected_scripts.length == 1 && scripts.include?(detected_scripts.first)
+        script = detected_scripts.first
+      end
+      script
+    end
+
+    def script_from_field(field)
+      scripts = region.combined_address_format.except("default").keys
+      script = scripts.find do |script|
+        combined_address_format = region.combined_address_format.dig(script, field)
+        additional_fields_string = combined_address_format.map { |field| send(field["key"].underscore.to_sym) }.join(" ")
+        text_exclusively_uses_script?(additional_fields_string, script)
+      end
+      script.present? ? script : "default"
+    end
```

Comment:
👍 Looks good, matches the Typescript logic:
1. Loop over each script found in the format
2. Analyze all fields and get a list of all scripts found across all fields (done here slightly differently by string joining but same idea)
3. If exactly one matching script is found, use that script, otherwise use "default"